### PR TITLE
ed, kraken: Rename frame in ed and kraken

### PR DESCRIPTION
**Pull request by kadhikari**

### CODE REVIEW

File: source/ed/connectors/fusio_parser.cpp [hunk 1]
```diff
@@ -46,54 +46,54 @@ typename C::mapped_type get_object(const C& map, const std::string& obj_id, cons
...
+Add default dataset for all contributor without any dataset
 */
-static void default_frames(GtfsData& gdata, Data& data){
+static void default_datasets(GtfsData& gdata, Data& data){
     for(auto* contributor: data.contributors){
         bool exist = false;
-        for(const auto frame : data.frames){
-            if (frame->contributor->uri == contributor->uri){
+        for(const auto dataset : data.datasets){
+            if (dataset->contributor->uri == contributor->uri){
```
Comment (antoine-de) [inline source/ed/connectors/fusio_parser.cpp:55]: it's not the issue here, but you could just return there and remove the `exist`

File: source/ed/connectors/fusio_parser.h [hunk 2]
```diff
@@ -161,6 +162,15 @@ struct FrameFusioHandler : public GenericHandler {
         "frame_start_date", "frame_end_date"}; }
 };
 
+struct DatasetFusioHandler : public GenericHandler {
+    DatasetFusioHandler(GtfsData& gdata, CsvReader& reader) : GenericHandler(gdata, reader) {}
+    int id_c, contributor_c, start_date_c, end_date_c, type_c, desc_c, system_c;
+    void init(Data&);
+    void handle_line(Data& data, const csv_row& row, bool is_first_line);
+    const std::vector<std::string> required_headers() const { return {"dataset_id", "contributor_id",
```
Comment (antoine-de) [inline source/ed/connectors/fusio_parser.h:170]: I think it's easier to only use the DatasetFusioHandler and remove the FrameFusioHandler no ?

File: source/sql/alembic/versions/3dc7d057ec3_rename_frame_to_dataset.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from alembic import op
+import sqlalchemy as sa
+import geoalchemy2 as ga
+
+
+def upgrade():
+    op.rename_table("frame", "dataset", schema='navitia')
+    op.add_column('vehicle_journey', sa.Column('dataset_id', sa.BIGINT(), nullable=True), schema='navitia')
+    op.execute("UPDATE navitia.vehicle_journey SET dataset_id=frame_id")
+    op.drop_column('vehicle_journey', 'frame_id', schema='navitia')
```
Comment (antoine-de) [inline source/sql/alembic/versions/3dc7d057ec3_rename_frame_to_dataset.py:24]: don't you have to also add the foreign key  ?

File: source/sql/alembic/versions/3dc7d057ec3_rename_frame_to_dataset.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    pass
```
Comment (antoine-de) [inline source/sql/alembic/versions/3dc7d057ec3_rename_frame_to_dataset.py:25]: You can remove the pass (and also in the dowgrade)

**Comment by pbougue** [comment]
Ok for me.
TODO:
- open a PR for after migration
- make a pass on this PR to add cosmetic corrections in the next PR

**Pull request merged**